src/app/shell/admin-tools/data/import-export/import-providers/import-providers.component.ts

```diff
@@ -0,0 +1,223 @@
+  }
+
+  public getProvidersData(wb: XLSX.WorkBook, wsname: string): IProviders[] {
+    return XLSX.utils.sheet_to_json(wb.Sheets[wsname], {
+      header: ['providerName', 'ownership', 'identifier', 'licenseNumber', 'settlement', 'address', 'email', 'phoneNumber'],
+      range: 1
+    });
+  }
+
+  public processProvidersData(providers: IProviders[]): void {
+    const isCorrectLength = this.cutArrayToHundred(providers);
+    const setIdToProviders: IProvidersID[] = providers.map((elem, index) => ({ ...elem, id: index }));
+    this.verifyEmailsEdrpous(setIdToProviders).subscribe((emailsEdrpous) => {
+      this.handleData(emailsEdrpous, setIdToProviders, isCorrectLength);
+    });
+  }
+
+  public handleData(emailsEdrpous: IEmailsEdrpousResponse, providers: IProvidersID[], isCorrectLength: boolean): void {
+    this.importValidationService.checkForInvalidData(providers, emailsEdrpous);
+    this.dataSource = providers;
+    this.dataSourceInvalid = this.filterInvalidProviders(providers);
+    this.isWaiting = false;
+    this.isWarningVisible = isCorrectLength;
+  }
+  // -----------------------------------------------------------------
+  public onFileSelected(event: any): void {
+    this.selectedFile = event.target.files[0];
+    this.isWaiting = true;
+    this.resetValues();
+    this.convertExcelToJSON(event);
+    event.target.value = '';
+  }
+
+  public verifyEmailsEdrpous(providers: IProvidersID[]): Observable<IEmailsEdrpousResponse> {
+    const emailsEdrpous: IEmailsEdrpous = providers.reduce(
+      (acc, element) => {
+        if (element.identifier && EDRPOU_IPN_REGEX.test(element.identifier.toString())) {
+          acc.edrpous[element.id] = element.identifier;
+        }
+        if (element.email && EMAIL_REGEX.test(element.email)) {
+          acc.emails[element.id] = element.email;
+        }
+        return acc;
+      },
+      { edrpous: {}, emails: {} }
+    );
+    return this.importService.sendEmailsEDRPOUsForVerification(emailsEdrpous);
+  }
+
+  public filterInvalidProviders(providers: IProvidersID[]): any {
+    return providers.filter((elem) => Object.values(elem.errors).find((error) => error !== null));
+  }
+  public checkHeadersIsValid(currentHeaders: string[]): boolean {
+    const standardHeaders = [
+      'Назва закладу',
+      'Форма власності',
+      'ЄДРПОУ',
+      'Ліцензія №',
+      'Населений пункт',
+      'Адреса',
+      'Електронна пошта',
+      'Телефон'
+    ];
+    const isValid = standardHeaders.every((header, index) => currentHeaders[index].trim() === header);
+    if (!isValid) {
+      this.isWaiting = false;
+      const invalidHeader = currentHeaders.find((header, index) => header !== standardHeaders[index]);
+      alert(`Невідповідність в заголовку "${invalidHeader}",\n\nЗразок:\n${standardHeaders.join(' | ')}`);
+    }
+    return isValid;
+  }
+
+  public cutArrayToHundred(providers: IProviders[]): boolean {
```

Comment:
Based on the name of the method, it looks like it should return an array, not a boolean value.
Rename it, or change logic